contracts/contracts/WhitelistPlugin.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+import {ISafe} from "./interfaces/Accounts.sol";
+import {ISafeProtocolPlugin} from "./interfaces/Integrations.sol";
+import {ISafeProtocolManager} from "./interfaces/Manager.sol";
+import {SafeTransaction, SafeRootAccess} from "./DataTypes.sol";
+import {BasePlugin, PluginMetaData} from "./Plugins.sol";
+
+/**
+ * @title WhitelistPlugin maintains a mapping that stores information about accounts that are
+ *        permitted to execute non-root transactions through a Safe account.
+ * @notice This plugin does not need Safe owner(s) confirmation(s) to execute Safe txs once enabled
+ *         through a Safe{Core} Protocol Manager.
+ */
+contract WhitelistPlugin is BasePlugin {
+    // safe account => account => whitelist status
+    mapping(address => mapping(address => bool)) public whitelistedAddresses;
+
+    event AddressWhitelisted(address indexed account);
+    event AddressRemovedFromWhitelist(address indexed account);
+
+    error AddressNotWhiteListed(address account);
+
+    constructor()
+        BasePlugin(PluginMetaData({name: "Whitelist Plugin", version: "1.0.0", requiresRootAccess: false, iconUrl: "", appUrl: ""}))
+    {}
+
+    /**
+     * @notice Executes a Safe transaction if the caller is whitelisted for the given Safe account.
+     * @param manager Address of the Safe{Core} Protocol Manager.
+     * @param safe Safe account
+     * @param safetx SafeTransaction to be executed
+     */
+    function executeFromPlugin(
+        ISafeProtocolManager manager,
```

Comment:
Should the manager be part of the constructor?

Reply:
Actually not sure :thinking: worth evaluating what the best approach is.

Reply:
For initial setup, where Manager gets updated frequently, I think it is fine to keep it as a parameter just for convenience.